Add redirector to proxy to the new sippy deployment on DPCR cluster.

dgoodwin opened the pull request.

dgoodwin commented:
/hold 

Need to contact @bparees  to manually apply these so we can test.

hongkailiu commented on clusters/app.ci/sippy-redirector/README.md:
```diff
@@ -0,0 +1,8 @@
+sippy-redirector
+====
+
+Deploys https://github.com/cgwalters/rhcos-redirector in app.ci.
```
```suggestion
Deploys a reverse proxy on `app.ci` which redirects traffic from https://sippy.ci.openshift.org/ to https://sippy.dptools.openshift.org/
```

hongkailiu reviewed (commented):
You will need this to direct the traffic to the redirector instead of the real `sippy` deployment.

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    cert-manager.io/cluster-issuer: cert-issuer
    haproxy.router.openshift.io/timeout: 2m
  name: sippy
  namespace: bparees
spec:
  rules:
  - host: sippy.ci.openshift.org
    http:
      paths:
      - backend:
          service:
            name: sippy-redirector
            port:
              number: 8080
        pathType: ImplementationSpecific
      - backend:
          service:
            name: metrics
            port:
              number: 2112
        pathType: ImplementationSpecific
  tls:
  - hosts:
    - sippy.ci.openshift.org
    secretName: sippy-tls

```

Do you mind updating the link in ci-docs before merging this one?
https://docs.ci.openshift.org/docs/getting-started/useful-links/#services

dgoodwin commented:
@hongkailiu what specific update were you thinking of? The link should still be valid. Did you just mean to remove sippy from there as it's no longer a ci service per se? Should I replace it with the fact that there's a sippy redirect service?

I've updated the readme and added the ingress definition based on what bparees setup when we did the migration, so it should match what's deployed now. IMO this PR and any potential docs pr should be able to merge in either order.

dgoodwin commented:
The changes are manually applied and working now.

hongkailiu commented:
/lgtm

> Did you just mean to remove sippy from there as it's no longer a ci service per se?

No. I meant to replace https://sippy.ci.openshift.org/ with https://sippy.dptools.openshift.org/.
I believe the latter is the one we encourage people to use since it is more direct, not replying on CI.

> IMO this PR and any potential docs pr should be able to merge in either order.
Since the objects have been verified, any order is fine. I was thinking we should ensure at least the one in the doc to work in the transactional phase.


It seems the original deployment has been removed. Cool.

```yaml
$ oc get deploy -n bparees
NAME               READY   UP-TO-DATE   AVAILABLE   AGE
sippy-redirector   2/2     2            2           3h11m
```

dgoodwin commented:
/hold cancel

@hongkailiu how strongly do you feel on the hostname? There must be hundreds of sippy links floating around in slack/bugzilla/jira, and I kinda assumed sippy.ci.openshift.org was our official long term hostname for that reason. We could undertake changing it wherever we see it and phase it out slowly but it's as pretty large task, I'd almost rather keep everyone on the one they're using and if things really got bad, we could do the dns zone solution you mentioned to move it if we can't redirect it anymore or that is causing problems.

I'm open to the alternative if you feel it's better.

hongkailiu commented:
> I kinda assumed sippy.ci.openshift.org was our official long term hostname for that reason. ... I'd almost rather keep everyone on the one they're using

I am totally fine if you want to keep the `sippy.ci.openshift.org` working, even long term if that is what you want.
I thought you want to start to phase it out right away.

dgoodwin commented:
Ok cool, thanks!
